crates/state/src/template_storage.rs

```diff
@@ -0,0 +1,65 @@
+use svm_types::{BytesPrimitive, Sections, TemplateAddr};
+
+use crate::{GlobalState, StorageResult};
+
+/// A [`GlobalState`] wrapper that exposes getters and setters for
+/// [`Template`](svm_types::Template)-related entities.
+pub struct TemplateStorage {
+    gs: GlobalState,
+    template_addr: TemplateAddr,
+}
+
+impl TemplateStorage {
+    /// Creates a new [`TemplateStorage`] from the given address and
+    /// [`GlobalState`] instance.
+    pub fn new(template_addr: &TemplateAddr, gs: GlobalState) -> Self {
+        Self {
+            template_addr: template_addr.clone(),
+            gs,
+        }
+    }
+
+    /// Reads, decodes and finally returns all [`Sections`] of `self`.
+    pub fn sections(&self) -> StorageResult<Option<Sections>> {
+        let core_sections_opt: Option<Sections> =
```

Comment:
can you please omit the `_opt` suffix?

Reply:
Done, it's better now. Sorry about doing this again, I lost this change after the last review.